api/src/main/java/org/pmiops/workbench/api/AuditController.java

```diff
@@ -0,0 +1,138 @@
+        .toFormatter();
+
+  private final Clock clock;
+  private final BigQueryService bigQueryService;
+  private final UserDao userDao;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  @Autowired
+  AuditController(
+      Clock clock,
+      BigQueryService bigQueryService,
+      UserDao userDao,
+      Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.clock = clock;
+    this.bigQueryService = bigQueryService;
+    this.userDao = userDao;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+  @VisibleForTesting
+  static String auditTableSuffix(Instant now, int daysAgo) {
+    Instant target = now.minus(daysAgo, ChronoUnit.DAYS);
+    return auditTableNameDateFormatter.withZone(ZoneId.of("UTC")).format(target);
+  }
+
+  private static String auditSql(String cdrProjectId, List<String> tableSuffixes) {
+    // "jobInsertResponse" appears to always be included, despite whether or not job request
+    // metadata was included (i.e. for jobs running in other projects).
+    String tableWildcard = String.format(
+        "%s.%s.cloudaudit_googleapis_com_data_access_*", cdrProjectId, AUDIT_SINK_NAME);
+    return String.format(
+        "SELECT\n" +
+        "  protopayload_auditlog.servicedata_v1_bigquery.jobInsertResponse.resource.jobName.projectId client_project_id,\n" +
+        "  protopayload_auditlog.authenticationInfo.principalEmail user_email,\n" +
+        "  SUM(1) total\n" +
+        "FROM `%s`\n" +
+        "WHERE protopayload_auditlog.methodName = 'jobservice.insert'\n" +
+        "  AND _TABLE_SUFFIX IN (%s)\n" +
+        "GROUP BY 1, 2",
+        tableWildcard,
+        tableSuffixes.stream().map(s -> "'" + s + "'").collect(Collectors.joining(",")));
+  }
+
+  @AuthorityRequired({Authority.AUDIT})
+  @Override
+  public ResponseEntity<AuditBigQueryResponse> auditBigQuery() {
+    // We expect to only see queries run within Firecloud AoU projects, or for administrative
+    // purposes within the CDR project itself.
+    String cdrProjectId = workbenchConfigProvider.get().server.projectId;
+    Set<String> whitelist = Sets.union(userDao.getAllUserProjects(), ImmutableSet.of(cdrProjectId));
+
+    Instant now = clock.instant();
+    List<String> suffixes = IntStream.range(0, AUDIT_DAY_RANGE)
+        .mapToObj(i -> auditTableSuffix(now, i))
+        .collect(Collectors.toList());
+
+    QueryResult result = bigQueryService.executeQuery(
+        QueryJobConfiguration.of(auditSql(cdrProjectId, suffixes)));
+    Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+
+    int numBad = 0;
+    int numQueries = 0;
+    for (List<FieldValue> row : result.iterateAll()) {
+      String project_id = bigQueryService.getString(row, rm.get("client_project_id"));
+      String email = bigQueryService.getString(row, rm.get("user_email"));
+      long total = bigQueryService.getLong(row, rm.get("total"));
+      if (bigQueryService.isNull(row, rm.get("client_project_id"))) {
```

Comment:
Will the audit logs contain requests to BQ that failed? Or only ones that succeeded?

(If it included failed ones, it seems that this would log cases where people tried to access BQ but couldn't.)

Reply:
Actually, just reread your comment and realized I missed your point, let me test this.

Reply:
Good catch. Permission denied requests were getting picked up here which was undesirable. Modified the query to filter out those cases.